### PR TITLE
parsing secondary_engine

### PR DESCRIPTION
**Pull request by JCOR11599**
This PR parses the syntax for `SECONDARY_ENGINE`, but does not store it with the table.
For the most part we seem to parse, but not store any table options.

MySQL docs aren't up-to-date yet; `SECONDARY_ENGINE` is not listed as  a valid table option, but it works.
https://dev.mysql.com/doc/refman/8.0/en/create-table.html

These are also apparently valid options for create index statements, so I just added the syntax there too
https://dev.mysql.com/doc/refman/8.0/en/create-index.html

Fix for: https://github.com/dolthub/dolt/issues/6632

### CODE REVIEW

File: go/vt/sqlparser/sql.y [hunk 1]
```diff
@@ -3740,6 +3740,14 @@ index_option:
   {
     $$ = &IndexOption{Name: string($1), Value: NewStrVal($2)}
   }
+| ENGINE_ATTRIBUTE equal_opt ID
```
Comment (max-hoffman) [inline go/vt/sqlparser/sql.y:3743]: test for index option? also does this one only support ID?

File: go/vt/sqlparser/sql.y [hunk 2]
```diff
@@ -4541,6 +4557,18 @@ alter_table_statement_part:
...
   }
+| SECONDARY_ENGINE equal_opt ID
+  {
+    $$ = &DDL{Action: AlterStr}
+  }
+| SECONDARY_ENGINE equal_opt STRING
+  {
+    $$ = &DDL{Action: AlterStr}
+  }
+| SECONDARY_ENGINE_ATTRIBUTE equal_opt STRING
```
Comment (max-hoffman) [inline go/vt/sqlparser/sql.y:4568]: only STRING?

**Review by max-hoffman — APPROVED** [review]
lgtm, doing one sanity pass to see if we're missing any other small obvious things from the docs wouldn't hurt

File: go/vt/sqlparser/sql.y (hunk 1 above)
Comment (JCOR11599) [reply go/vt/sqlparser/sql.y:3743]: Oops this is supposed to be only STRING, not ID.

File: go/vt/sqlparser/sql.y (hunk 2 above)
Comment (JCOR11599) [reply go/vt/sqlparser/sql.y:4568]: Weirdly, yeah. `SECONDARY_ENGINE` can do both, but not `SECONDARY_ENGINE_ATTRIBUTE`. They seem to be different table options.

**Pull request merged**